tooling/minimize/src/util.rs

```diff
@@ -0,0 +1,132 @@
+use std::env;
+use std::ffi::OsStr;
+
+use std::path::PathBuf;
+use std::process::Command;
+
+use rustc_build_sysroot::{BuildMode, SysrootBuilder, SysrootConfig};
+
+const DEFAULT_ARGS: &[&str] = &[
+    // This is the same as Miri's `MIRI_DEFAULT_ARGS`, ensuring we get a MIR with all the UB still present.
+    "--cfg=miri",
+    "-Zalways-encode-mir",
+    "-Zextra-const-ub-checks",
+    "-Zmir-emit-retag",
+    "-Zmir-opt-level=0",
+    "-Zmir-enable-passes=-CheckAlignment",
+    "-Zmir-keep-place-mention",
+    // Also disable UB checks (since `cfg(miri)` in the standard library do not trigger for us).
+    "-Zub-checks=false",
+];
+
+pub fn insert_default_args(args: &mut Vec<String>, index: usize) {
+    args.splice(index..index, DEFAULT_ARGS.iter().map(ToString::to_string));
+    if std::env::var("MINIMIZE_BE_RUSTC").as_deref() != Ok("sysroot") {
+        let sysroot = get_sysroot_dir();
+        args.insert(index, format!("--sysroot={}", sysroot.display()));
+    }
+}
+
+pub fn show_error(msg: &impl std::fmt::Display) -> ! {
+    eprintln!("fatal error: {msg}");
+    std::process::exit(101) // exit code needed to make ui_test happy
+}
+
+#[macro_export]
+macro_rules! show_error {
+    ($($tt:tt)*) => {show_error(&format_args!($($tt)*)) };
+}
+
+pub fn get_sysroot_dir() -> PathBuf {
+    match std::env::var_os("MINIRUST_SYSROOT") {
+        Some(dir) => PathBuf::from(dir),
+        None => {
+            let user_dirs = directories::ProjectDirs::from("org", "rust-lang", "minirust").unwrap();
+            user_dirs.cache_dir().to_owned()
+        }
+    }
+}
+
+pub fn be_rustc() {
+    // Get the rest of the command line arguments
+    let mut args: Vec<String> = env::args().skip(1).collect();
+
+    let use_panic_abort = args
+        .array_windows::<2>()
+        .any(|[first, second]| first == "--crate-name" && second == "panic_abort");
+
+    insert_default_args(&mut args, 0);
+
+    if use_panic_abort {
+        args.insert(0, "-Cpanic=abort".into());
+    } else {
+        args.insert(0, "-Cpanic=unwind".into());
+    }
+
+    // Invoke the rust compiler
+    let status = Command::new("rustc")
+        .args(args)
+        .env_remove("RUSTC")
+        .env_remove("MINIMIZE_BE_RUSTC")
+        .status()
+        .expect("failed to invoke rustc in custom sysroot build");
+
+    std::process::exit(status.code().unwrap_or(1));
```

Comment:
That is not what we discussed about how minimize can be rustc (we discussed doing it like Miri, which does not invoke `rustc` as a binary); now you rely on this `rustc` having the exact right version for everything to match.

But I guess it works and we already relied on it before so we can also do it like this. I would just like you to tell me when you decide to use a different approach than what we discussed.